File: sem3/map/Lab/toylanguage_C#/ToyLanguage/ToyLanguage/Models/Statements/AssignmentStatement.cs
using ToyLanguage.Models.Interfaces;
using ToyLanguage.Models.States;

namespace ToyLanguage.Models.Statements
{
    public class AssignmentStatement : IStatement
    {
        private readonly string _variableName;
        private readonly IExpression _expression;

        public AssignmentStatement(string variableName, IExpression expression)
        {
            _variableName = variableName;
            _expression = expression;
        }
        
        public ProgramState Execute(ProgramState state)
        {
            state.SymbolTable[_variableName] = _expression.Eval(state);
            
            return state;
        }

        public override string ToString()
        {
            return $"{_variableName} = {_expression}";
        }
    }
}

File: sem3/map/Exam/GUI_ToyLanguageInterpreter/src/Interpreter/Models/Statements/AssignmentStatement.java
package Interpreter.Models.Statements;

import Interpreter.Models.Expression;
import Interpreter.Models.IStatement;
import Interpreter.Models.ProgramState.ProgramState;
import Interpreter.Utils.MyIDictionary;

public class AssignmentStatement implements IStatement {
    private String variableName;
    private Expression expression;

    public AssignmentStatement(String variableName, Expression expression) {
        this.variableName = variableName;
        this.expression = expression;
    }

    @Override
    public ProgramState execute(ProgramState state) throws Exception {
        MyIDictionary<String, Integer> symTable = state.getSymTable();

        int expressionValue = this.expression.eval(state);

        if (symTable.contains(this.variableName)) {
            symTable.setValue(this.variableName, expressionValue);
        } else {
            symTable.add(this.variableName, expressionValue);
        }

        return null;
    }

    @Override
    public String toString() {
        return this.variableName + " = " + this.expression.toString();
    }
}
